lib/Infer/AliveDriver.cpp

```diff
@@ -0,0 +1,401 @@
+  }
+
+  template <typename A, typename B>
+  IR::Value *iCmp(IR::Type &t, std::string name,
+                  IR::ICmp::Cond cond, A a, B b) {
+    return append
+      (std::make_unique<IR::ICmp>
+      (t, std::move(name), cond, *toValue(t, a), *toValue(t, b)));
+  }
+
+  IR::Value *var(IR::Type &t, std::string name) {
+    return toValue(t, name);
+  }
+
+  IR::Value *val(IR::Type &t, int64_t val) {
+    return toValue(t, val);
+  }
+
+  template<typename T>
+  IR::Value *ret(IR::Type &t, T ret) {
+    return append(
+      std::make_unique<IR::Return>(t, *toValue(t, ret)));
+  }
+
+  template <typename T>
+  void assume(T &&V) {
+    auto AI = std::make_unique<IR::Assume>(*std::move(V));
+    F.getBB("").addIntr(std::move(AI));
+  }
+
+  // Unimplemented : Freeze, CopyOp, Unreachable
+
+private:
+  template <typename T>
+  IR::Value *append(T &&p) {
+    auto ptr = p.get();
+    F.getBB("").addIntr(std::move(p));
+    return ptr;
+  }
+
+  IR::Function &F;
+  std::unordered_map<std::string, IR::Value *> identifiers;
+
+  IR::Value *toValue(IR::Type &t, int64_t x) {
+    auto c = std::make_unique<IR::IntConst>(t, x);
+    auto ptr = c.get();
+    F.addConstant(std::move(c));
+    return ptr;
+  }
+
+  IR::Value *toValue(IR::Type &t, std::string x) {
+    if (auto It = identifiers.find(x); It != identifiers.end()) {
+      return It->second;
+    } else {
+      auto i = std::make_unique<IR::Input>(t, std::move(x));
+      auto ptr = i.get();
+      F.addInput(std::move(i));
+      identifiers[x] = ptr;
+      return ptr;
+    }
+  }
+
+  IR::Value *toValue(IR::Type &t, const char *x) {
+    return toValue(t, std::string(x));
+  }
+
+  IR::Value *toValue(IR::Type &, IR::Value *v) {
+    return v;
+  }
+};
+
+
+bool startsWith(const std::string &pre, const std::string &str) {
+  return std::equal(pre.begin(), pre.end(), str.begin());
+}
+
+void getReservedConsts(souper::Inst *I, std::set<std::string> &Result) {
+  if (startsWith("reservedconst_", I->Name)) {
+    Result.insert(I->Name);
+  }
+  for (auto &&Op : I->Ops) {
+    getReservedConsts(Op, Result);
+  }
+}
+
+struct ReturnLHSRAII {
+  tools::Transform &t;
+  IR::Function &LHS;
+  ~ReturnLHSRAII() {
+    LHS = std::move(t.src);
+  }
+};
+}
+
+util::Errors synthesizeConstantUsingSolver(tools::Transform &t,
+                                           int64_t &result) {
+
+  IR::Value::reset_gbl_id();
+  IR::State src_state(t.src), tgt_state(t.tgt);
+  util::sym_exec(src_state);
+  util::sym_exec(tgt_state);
+
+  util::Errors errs;
+
+  auto &ap = src_state.returnVal();
+  auto &bp = tgt_state.returnVal();
+  auto &a = ap.first;
+  auto &b = bp.first;
+
+  auto qvars = src_state.getQuantVars();
+  qvars.insert(ap.second.begin(), ap.second.end());
+
+  auto err = [&](const smt::Result &r, bool print_var, const char *msg) {
+    tools::error(errs, src_state, tgt_state, r, print_var, nullptr, a, b, msg,
+          false);
+  };
+
+  std::set<smt::expr> vars;
```

Comment:
please follow the LLVM coding style for local variable names https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly